include/nbl/builtin/hlsl/scan/virtual_workgroup.hlsl

```diff
@@ -6,87 +6,78 @@
 #include "nbl/builtin/hlsl/math/typeless_arithmetic.hlsl"
 #include "nbl/builtin/hlsl/workgroup/arithmetic.hlsl" // This is where all the nbl_glsl_workgroupOPs are defined
 #include "nbl/builtin/hlsl/scan/declarations.hlsl"
-
+#include "nbl/builtin/hlsl/scan/default_scheduler.hlsl"
 #include "nbl/builtin/hlsl/binops.hlsl"
 
-#if 0
 namespace nbl
 {
 namespace hlsl
 {
 namespace scan
 {
-	template<class Binop, class Storage_t>
-	void virtualWorkgroup(in uint treeLevel, in uint localWorkgroupIndex)
+	template<class Binop, typename Storage_t, bool isExclusive, uint16_t ItemCount, class Accessor, class device_capabilities=void>
```

Comment:
why not template on the workgroup scan instead?

Within you can alias and extract:
- binop
- storage_t
- exclusive or not
- item count
- smem accessor
- device traits necessary